### ISSUE TITLE
Hi~How should I infer a video by using super-gradients？

### ISSUE THREAD

**Issue by AddPear** [Deci-AI/super-gradients#1156]
### 💡 Your Question

from super_gradients.common.object_names import Models
from super_gradients.training import models

if __name__ == '__main__':
    # init_trainer()
    # setup_device("cuda")
    # Load the best model that we trained
    from super_gradients.training.processing import DetectionCenterPadding, StandardizeImage, NormalizeImage, \
        ImagePermute, ComposeProcessing, DetectionLongestMaxSizeRescale

    image_processor = ComposeProcessing(
        [
            DetectionLongestMaxSizeRescale(output_shape=(636, 636)),
            DetectionCenterPadding(output_shape=(640, 640), pad_value=114),
            StandardizeImage(max_value=255.0),
            ImagePermute(permutation=(2, 0, 1)),
        ]
    )
    yolo_nas = models.get(Models.YOLO_NAS_S, num_classes=5, checkpoint_path="ckpt_best.pth").cuda()

    class_names = [str(i) for i in range(5)]
    yolo_nas.set_dataset_processing_params(
        class_names=class_names,
        image_processor=image_processor,
        iou=0.35, conf=0.25,
    )

    # yolo_nas.predict('1.jpg', ).show()
    yolo_nas.predict("test.mp4").show()

# When I ran this code, my computer crashed

### Versions

D:\Location_Programming\Anaconda3\envs\yolonas\python.exe F:/_code/__ketang-biye/super-gradients-master/collect_env.py 
Collecting environment information...
PyTorch version: 1.9.1+cu111
Is debug build: False
CUDA used to build PyTorch: 11.1
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 Pro
GCC version: (x86_64-win32-seh-rev0, Built by MinGW-W64 project) 8.1.0
Clang version: Could not collect
CMake version: version 3.17.20200511-g8787079
Libc version: N/A

Python version: 3.9.16 (main, Mar  8 2023, 10:39:24) [MSC v.1916 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.22621-SP0
Is CUDA available: True
CUDA runtime version: 11.1.74
CUDA_MODULE_LOADING set to: 
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3060
Nvidia driver version: 517.48
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:

Revision=20480

Versions of relevant libraries:
[pip3] numpy==1.23.0
[pip3] torch==1.9.1+cu111
[pip3] torchaudio==0.9.1
[pip3] torchmetrics==0.8.0
[pip3] torchvision==0.10.1+cu111
[conda] blas                      1.0                         mkl    https://mirrors.ustc.edu.cn/anaconda/pkgs/main
[conda] mkl                       2023.1.0         h8bd8f75_46356    https://mirrors.ustc.edu.cn/anaconda/pkgs/main
[conda] mkl-service               2.4.0            py39h2bbff1b_1    https://mirrors.ustc.edu.cn/anaconda/pkgs/main
[conda] mkl_fft                   1.3.6            py39hf11a4ad_1    https://mirrors.ustc.edu.cn/anaconda/pkgs/main
[conda] mkl_random                1.2.2            py39hf11a4ad_1    https://mirrors.ustc.edu.cn/anaconda/pkgs/main
[conda] numpy                     1.23.0                   pypi_0    pypi
[conda] pytorch-cuda              11.7                 h16d0643_5    pytorch
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] torch                     1.9.1+cu111              pypi_0    pypi
[conda] torchaudio                0.9.1                    pypi_0    pypi
[conda] torchmetrics              0.8.0                    pypi_0    pypi
[conda] torchvision               0.10.1+cu111             pypi_0    pypi

Process finished with exit code 0

**Comment by BloodAxe** [comment]
Since 3.4 we have optimized video prediction so now `model.predict("long_video.mp4").save("output.mp4")` should not OOM.

**Comment by CrasCris** [comment]
> Since 3.4 we have optimized video prediction so now `model.predict("long_video.mp4").save("output.mp4")` should not OOM.

it's possible to inference using live video with my webcam computer

**Comment by BloodAxe** [comment]
Yes, you can use `model.predict_webcam()` method for this
